src/student_success_tool/modeling/evaluation.py

```diff
@@ -64,6 +65,144 @@ def extract_training_data_from_model(
     return df_loaded
 
 
+def load_model_and_predict(
+    experiment_id: str,
+    run_id: str,
+    optimization_metric: str,
+    df: pd.DataFrame,
+    df_features: pd.DataFrame,
+    pred_col: str,
+    pred_prob_col: str,
+) -> tuple[sklearn.base.BaseEstimator, pd.DataFrame]:
+    """
+    Loads a trained MLflow model, generates predictions, logs a model comparison figure,
+    and returns a DataFrame containing predictions.
+
+    Args:
+        experiment_id (str): The MLflow experiment ID containing the trained models.
+        run_id (str): The specific MLflow run ID to load the model from.
+        optimization_metric (str): The metric used to compare and rank trained models.
+        df (pd.DataFrame): The original DataFrame containing all relevant columns.
+        df_features (pd.DataFrame): DataFrame but with only columns from model training features.
+        pred_col (str): Column name for the predictions.
+        pred_prob_col (str): Column name for predicted probabilities.
+
+    Returns:
+        model (sklearn.base.BaseEstimator): The loaded model from experiment_id/run_id.
+        df_pred (pd.DataFrame): A copy of the original DataFrame `df`, with two new columns: pred_col: predicted
+        class labels and pred_probs_col: predicted probabilities for the positive class.
+    """
+    model = mlflow.sklearn.load_model(f"runs:/{run_id}/model")
+    df_pred = df.assign(
+        **{
+            pred_col: model.predict(df_features),
+            pred_prob_col: model.predict_proba(df_features)[:, 1],
+        }
+    )
```

Comment:
We also have a function for this: https://github.com/datakind/student-success-tool/blob/develop/src/student_success_tool/modeling/inference.py#L14-L61

The logic here makes assumptions about `pos_label` that aren't made in other functions, so could easily get out of alignment if something were to change. Better to use the general-purpose existing function.